neo4j/exceptions.py

```diff
@@ -68,15 +68,47 @@
 """
 
 
+from __future__ import annotations
+
+import typing as t
+
+
+if t.TYPE_CHECKING:
+    import typing_extensions as te
+
+    from ._async.work import (
+        AsyncManagedTransaction,
+        AsyncResult,
+        AsyncSession,
+        AsyncTransaction,
+    )
+    from ._sync.work import (
+        ManagedTransaction,
+        Result,
+        Session,
+        Transaction,
+    )
+
+    _T_Transaction = t.Union[AsyncManagedTransaction, AsyncTransaction,
+                             ManagedTransaction, Transaction]
+    _T_Result = t.Union[AsyncResult, Result]
+    _T_Session = t.Union[AsyncSession, Session]
+else:
+    _T_Transaction = t.Union["AsyncManagedTransaction", "AsyncTransaction",
+                             "ManagedTransaction", "Transaction"]
+    _T_Result = t.Union["AsyncResult", "Result"]
+    _T_Session = t.Union["AsyncSession", "Session"]
```

Comment:
Because import `._async.work`/`._sync.work` causes a cyclic dependency here: :boom: 
So I use a forward reference (done with strings) if not type checking. This is actually only necessary to make the docs tool pick up the correct types when I use `_T_Transaction`, etc. as return/parameter type.

I could probably get away with only using forward referencing (in both cases) :shrug: